Ui telemetry statements details

nathanstilwell opened the pull request:
fixes #45625 

- [x] clicks of the sub navigation (logical plans, execution stats, diagnostics)
- [x] clicks of the diagnostics activation
- [x] click to activate modal on statements page

_Author's note_: It may be helpful to view this PR a commit at a time as there is some refactoring work in here, so I have tried to break the commits into logical steps that may tell a better story.

Organizing track functions into a consolidated location (`pkg/ui/src/util/analytics`) and writing them in a form that makes testing less cumbersome. The thrust of this PR is to add tracking for interesting events on the statement details page. This resolves two from issue #45625 and an additional event (identified from feedback) on the statements page. They are as follows,

### SubNavigation Selection

This event is fired when a user clicks on a tab in the sub-navigation.

![subnavigation-selection](https://user-images.githubusercontent.com/397448/78068152-ef2fcd80-7365-11ea-85f8-a7b4b55c68a7.gif)

```
{
  userId: '9553cdfb-5cf3-4430-862c-1a7bd4ec524b',
  event: 'SubNavigation Selection',
  properties: {
    pagePath: '/statements/(internal)/true/[some long internal fingerprint]',
    selection: 'logical-plan'
  }
}
```

### Diagnostics Activation

This event is fired when statement diagnostics are activated from the "Diagnostics" section of the details view,

![details-diagnostics-activation](https://user-images.githubusercontent.com/397448/78068320-2f8f4b80-7366-11ea-8852-653608d6fac4.gif)

```
{
  userId: '5f87309b-b68a-4c56-898f-4995e8672991',
  event: 'Diagnostics Activation',
  properties: {
    fingerprint: 'SELECT blah, blah FROM blahhdy-blah WHERE blah = $blah',
    pagePath: '/statement/[statement]'
  }
}
```
_and a backfill on the Statements view_

### Diagnostics Modal Open

This event is fired when the user clicks _"Activate"_ to open the activation modal.

![diagnostics-modal-open](https://user-images.githubusercontent.com/397448/78072603-6583fe00-736d-11ea-978f-dece08eb70e0.gif)
 
This event 
```
{
  userId: '9553cdfb-5cf3-4430-862c-1a7bd4ec524b',
  event: 'Diagnostics Modal Open',
  properties: {
    fingerprint: 'INSERT INTO test VALUES (_)',
    pagePath: '/statements'
  }
}
```

Commit:
ui: Added tracking for statement details sub-navigation

Added an analytics tracking call when the user interacts with the
sub-navigation in statement details. Since the sub-nav are tabs
implemented with AntD the only information provided to onChange is the
`key` of the TabPane, so to provide clearer analytics I changed the key
to something human readale while remaining unique.

Release justification: Low risk telemetry change

Release note (admin ui change): Add tracking for statement details
sub-navigation

Commit:
ui: Created a utility directory for tracking functions

There will be numerous tracking calls throughout the application and I
have started to already see reusable tracking functions across
components. To keep things orderly, I have started to consolidate the
functions into a single directory (`pkg/ui/src/util/analytics`).

In seperating them from the components where they are utilized, the
functions can become more testable. I have seperated each of these track
calls into a pure function that constructs an analytics event and calls
a track function, and then exported a default function that uses this
pure function passing it the analytics track method.

Release note: None

Commit:
ui: Uses extracted track call for subnav selection

Release note: None

Commit:
ui: Used extracted track call for diagnostics activation

Here was a case where the same tracking call was made across to
components. This method was moved to `util/analytics` and so I have
refactored it here.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46789)
<!-- Reviewable:end -->

Commit:
Add file headers for analytics tracking

Release note: None

Commit:
ui: added tracking when diagnostics modal opens

Added an additional track event on the statements page to see when the
user opens the modal to activate statement diagnostics.

Release note: None

Commit:
ui: refactored trackActivateDiagnostics test

Broke up the assertions to create more focused tests.

Release note: None

Commit:
ui: moved trackTableSort to util/analytics

Release note: None

Commit:
ui: move trackPaginate to util/analytics

Release note: None

Commit:
ui: move trackSearch to util/analytics

Release note: None

Commit:
ui: Code review changes

Giving these changes one last scan before I ask for review

Release note: None

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks for the detailed testing! I like that this sets a clear example for how future tracking impls should be done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46789#-:-M3qo0311Sf9y2p-sHbf:bxtysqb)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian and @koorosh)



<!-- Sent from Reviewable.io -->

nathanstilwell commented:
bors r=dhartunian

nathanstilwell commented:
bors r+

nathanstilwell commented:
bors retry
